### PR TITLE
ISPN-10690 Caches endpoint starting point

### PR DESCRIPTION
**Pull request by karesti**
Super simple endpoint, stating point for caches info page

https://issues.jboss.org/browse/ISPN-10690

### CODE REVIEW

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 1]
```diff
@@ -180,6 +184,43 @@ public Invocations getInvocations() {
       return completedFuture(responseBuilder.build());
    }
 
+   private CompletionStage<RestResponse> getCaches(RestRequest request) {
+      NettyRestResponse.Builder responseBuilder = checkCacheManager(request);
+      if (responseBuilder.getHttpStatus() == NOT_FOUND) return completedFuture(responseBuilder.build());
+
+      class CacheInfo {
```
Comment (gustavonalle) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: IMHO this class should be declared outside the method and also ```private static```

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 2]
```diff
@@ -180,6 +184,43 @@ public Invocations getInvocations() {
...
+            cacheInfo.name = cacheName;
+            cacheInfo.type = cache.getCacheConfiguration().clustering().cacheMode().toCacheType();
+            cacheInfo.status = cache.getStatus().name();
+            cacheInfo.size = cache.size();
+            cacheInfo.simpleCache = cache.getCacheConfiguration().simpleCache();
+            cacheInfo.transactional = cache.getCacheConfiguration().transaction().transactionMode().isTransactional();
+            cacheInfo.persistent = false;
+            return cacheInfo;
+         }).collect(Collectors.toSet());
+
```
Comment (gustavonalle) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: This calculation can block, so it should be wrapped in an executor

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 3]
```diff
@@ -91,6 +92,9 @@ public Invocations getInvocations() {
 
             // Stats
             .invocation().methods(GET).path("/v2/cache-managers/{name}/stats").handleWith(this::getStats)
+
+            // Caches
+            .invocation().methods(GET).path("/v2/cache-managers/{name}/caches").handleWith(this::getCaches)
```
Comment (gustavonalle) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: Were already have something similar in the ```CacheResourceV2```: https://github.com/infinispan/infinispan/blob/7ad546e0dfbe41a8155e83cf1bf6c4f811ab57f7/server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java#L69

I think you should move your logic there, maybe with a request parameter ```?detailed=true```

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 1 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: allright

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 2 above)
Comment (wburns) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: Are you sure this is blocking? This looks like it is just a CHM underneath.

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 3 above)
Comment (gustavonalle) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: Hmm...ok. But we need to unify those two...One returns only the names, this one the name + the details. What about bring the other one here?

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 2 above)
Comment (gustavonalle) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: cache.size isn't blocking?

Comment (wburns) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: Ah, yes. Sorry, I was looking at the other ones :)

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 3 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: but the other are all the caches... well now that we have a single cache manager both should bring the same amount of caches...

**Comment by karesti** [comment]
I pushed some changes

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 3 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: I agree we should decide how to do this 

@tristantarrant are you using this method in the CLI ? the caches one.

Comment (gustavonalle) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: Until we know, let's keep it here, and mark the API as 'experimental' in the docs :)

Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: you mean this new api method ?

Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: yep I did not change the docs

**Comment by karesti** [comment]
@gustavonalle added some doc

**Pull request merged**